packages/nestjs-zod/src/pipe.ts

```diff
@@ -1,40 +1,93 @@
-import { validate } from './validate'
+import { createZodValidationException, ZodExceptionCreator } from './exception'
 import { UnknownSchema } from './types'
+import { validate } from './validate'
 
 interface ZodValidationPipeOptions {
   createValidationException?: ZodExceptionCreator
 }
 
-type ZodValidationPipeClass = new (
-  schemaOrDto?: UnknownSchema | ZodDto<UnknownSchema>
-) => PipeTransform
+type ZodValidationPipeClass = {
+  new (options: ZodValidationPipeConstructorOptions): PipeTransform
+  new (schemaOrDto?: UnknownSchema | ZodDto<UnknownSchema>): PipeTransform
+}
+
+export interface ZodValidationPipeConstructorOptions {
+  schema?: UnknownSchema | ZodDto<UnknownSchema>
+  /**
+   * If this flag is set to true, inputs that do not specify a schema (either
+   * via the {@link ZodValidationPipeConstructorOptions.schema}, or by using a
+   * ZodDto class), will be rejected. This prevents unvalidated inputs from
+   * getting passed the {@link ZodValidationPipe}.
+   *
+   * @default false
+   */
+  rejectNoSchema?: boolean
+}
 
 export function createZodValidationPipe({
-  createValidationException,
+  createValidationException = createZodValidationException,
 }: ZodValidationPipeOptions = {}): ZodValidationPipeClass {
   @Injectable()
   class ZodValidationPipe implements PipeTransform {
-    constructor(private schemaOrDto?: UnknownSchema | ZodDto<UnknownSchema>) {}
+    private readonly schema: UnknownSchema | ZodDto<UnknownSchema> | undefined
+    private readonly rejectNoSchema?: boolean
+
+    constructor(options: ZodValidationPipeConstructorOptions)
+    constructor(schemaOrDto?: UnknownSchema | ZodDto<UnknownSchema>)
+    constructor(
+      schemaOrOptions:
+        | UnknownSchema
+        | ZodDto<UnknownSchema>
+        | ZodValidationPipeConstructorOptions = {}
+    ) {
+      if (isZodDto(schemaOrOptions) || 'parse' in schemaOrOptions) {
+        this.schema = schemaOrOptions
+      } else {
+        this.schema = schemaOrOptions.schema
+        this.rejectNoSchema = schemaOrOptions.rejectNoSchema
+      }
+    }
 
     public transform(value: unknown, metadata: ArgumentMetadata) {
-      if (this.schemaOrDto) {
-        return validate(value, this.schemaOrDto, createValidationException)
+      if (this.schema) {
+        return validate(value, this.schema, createValidationException)
       }
 
-      const { metatype } = metadata
+      const { metatype, type: paramType, data } = metadata
+
+      if (isZodDto(metatype)) {
+        return validate(value, metatype.schema, createValidationException)
+      }
 
-      if (!isZodDto(metatype)) {
+      if (!this.rejectNoSchema) {
         return value
       }
 
-      return validate(value, metatype.schema, createValidationException)
+      const runtimeType = PRIMITIVE_TYPES.get(metatype)
+      if (runtimeType && typeof value === runtimeType) {
+        // Allow simple `@Body() param: number`, `@Query('queryArg') query: string`, etc. cases.
+        return value
```

Comment:
I'm not sure I like this exception. I'm thinking of cases where the property should be declared as a Zod primitive schema (e.g. `z.string().meta({ description: "The user's name" })`) via [z.meta()](https://zod.dev/metadata?id=meta) in Zod-v4 to supply an OpenAPI spec description for a query param.

This exception may cause such DTOs to be missed and result in OpenAPI specs that are not fully described.

Reply:
@Simon-Hayden-Dev I've rebased the PR and fixed the conflicts, and also pushed a few tweaks.  Thanks for your contribution 🙏